benchmark/single-source/StringEdits.swift

```diff
@@ -31,14 +31,11 @@ var editWords: [String] = [
 let alphabet = "abcdefghijklmnopqrstuvwxyz"
 /// All edits that are one edit away from `word`
 func edits(_ word: String) -> Set<String> {
-  // create right/left splits as CharacterViews instead
   let splits = word.indices.map {
-    (word[..<$0], word[$0...])
+    (String(word[..<$0]), String(word[$0...]))
```

Comment:
@airspeedswift @moiseev It seems to me that this fundamentally changed the core type in this benchmark, as it no longer operates on `Substring` views, but creates `String` copies, as can be seen in the modified type of `result` below.

This also implicitly changes the argument types of the string concatenation functions in the main workload.

Even with this change, the last benchmark run here has shown a slight regression:

TEST	| OLD	| NEW	| DELTA	| SPEEDUP
--|--|--|--|--|
StringEdits	| 122997	| 140316	| +14.1%	| 0.88x

and @moiseev said:
> `StringEdits` is almost back to norm.

…but this isn't the same benchmark anymore. When I've switched locally to running with `Substring`s, it get's worse (from: 264 500 to 293 400), but we are now running on different string guts, so its not directly comparable.

What should this benchmark test: `String` or `Substring`? 
Should I add a second variant for comparison?

Reply:
> as it no longer operates on Substring views, but creates String copies

Because that's what it used to do before `Substring` was introduced. This change IIRC was made precisely because this was the minimal change to the benchmarks that would still allow us compare apples to apples, as opposed to rewriting a benchmark completely to use `Substring` as a common currency type.

I believe there is value in having both `String` and `Substring` based benchmarks, but @milseman would know for sure.

Reply:
> Because that's what it used to do before Substring was introduced. 

If you look at the code — [[View Changes]](https://github.com/apple/swift/pull/12196/files/53ec0b1168fade46f83a478f0f12780ce7e7e710#diff-b251a3b7fa95677d2852e04f4097a2d9L41) — the original type was `[String.CharacterView]` which I believe corresponds to `Substring`, right?

Reply:
Ping @milseman

Reply:
Ping @milseman, @airspeedswift